### PR TITLE
Realtime updates

### PR DESCRIPTION
**Pull request by gediminasel**

### CODE REVIEW

File: app/src/main/java/com/nmakademija/nmaakademija/api/listener/AcademicLoadedListener.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 public interface AcademicLoadedListener {
 
-    void onAcademicLoaded(Academic academic);
+    void onAcademicLoaded(Academic item);
```
Comment (vycius) [inline app/src/main/java/com/nmakademija/nmaakademija/api/listener/AcademicLoadedListener.java]: `academic`

**Review by kuprelis — APPROVED** [review]

File: app/src/main/java/com/nmakademija/nmaakademija/api/controllers/FirebaseController.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.google.firebase.database.ValueEventListener;
+
+public abstract class FirebaseController {
+    ValueEventListener eventListener;
+    DatabaseReference databaseReference;
+
+    public abstract void onCreate();
+
+    public void onDestroy() {
+        if (eventListener != null) {
```
Comment (vycius) [inline app/src/main/java/com/nmakademija/nmaakademija/api/controllers/FirebaseController.java]: How can it be null ?

File: app/src/main/java/com/nmakademija/nmaakademija/api/controllers/FirebaseController.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.nmakademija.nmaakademija.api.controllers;
+
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.ValueEventListener;
+
+public abstract class FirebaseController {
+    ValueEventListener eventListener;
+    DatabaseReference databaseReference;
+
+    public abstract void onCreate();
```
Comment (vycius) [inline app/src/main/java/com/nmakademija/nmaakademija/api/controllers/FirebaseController.java]: I don't like this method name.

Comment (vycius) [reply app/src/main/java/com/nmakademija/nmaakademija/api/controllers/FirebaseController.java]: It is slightly confusing as this is called from `onStart`.

**Review by vycius — APPROVED** [review]

**Pull request merged**